lib/recognizer/accounts.ex

```diff
@@ -162,13 +166,32 @@ defmodule Recognizer.Accounts do
       {:error, %Ecto.Changeset{}}
 
   """
-  def register_user(attrs, opts \\ []) do
-    if Map.get(attrs, "newsletter") == "true", do: Recognizer.Hal.update_newsletter(attrs)
+  def register_user(attrs, opts \\ [])
 
+  def register_user(attrs, verify_account_url_fun: verify_account_url_fun) do
+    %User{}
+    |> User.registration_changeset(attrs)
+    |> insert_user_and_notification_preferences()
+    |> maybe_generate_verification_code(verify_account_url_fun)
+    |> maybe_send_newsletter(attrs)
```

Comment:
All this "maybe" stuff should probably be replaced by a `with` statement, but it's interesting to write it this way.